{"expand":"renderedFields,names,schema,operations,editmeta,changelog,versionedRepresentations","id":"12517744","self":"https://issues.apache.org/jira/rest/api/2/issue/12517744","key":"HIVE-2338","fields":{"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/secure/viewavatar?size=xsmall&avatarId=21133&avatarType=issuetype","name":"Bug","subtask":false,"avatarId":21133},"timespent":null,"project":{"self":"https://issues.apache.org/jira/rest/api/2/project/12310843","id":"12310843","key":"HIVE","name":"Hive","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/projectavatar?pid=12310843&avatarId=11935","24x24":"https://issues.apache.org/jira/secure/projectavatar?size=small&pid=12310843&avatarId=11935","16x16":"https://issues.apache.org/jira/secure/projectavatar?size=xsmall&pid=12310843&avatarId=11935","32x32":"https://issues.apache.org/jira/secure/projectavatar?size=medium&pid=12310843&avatarId=11935"},"projectCategory":{"self":"https://issues.apache.org/jira/rest/api/2/projectCategory/10292","id":"10292","description":"Scalable Distributed Computing","name":"Hadoop"}},"fixVersions":[{"self":"https://issues.apache.org/jira/rest/api/2/version/12316178","id":"12316178","description":"released","name":"0.8.0","archived":false,"released":true,"releaseDate":"2011-12-16"}],"aggregatetimespent":null,"resolution":{"self":"https://issues.apache.org/jira/rest/api/2/resolution/1","id":"1","description":"A fix for this issue is checked into the tree and tested.","name":"Fixed"},"customfield_12312322":null,"customfield_12310220":"2011-08-03T04:13:27.623+0000","customfield_12312520":null,"customfield_12312323":null,"customfield_12312521":"Tue Aug 23 12:33:23 UTC 2011","customfield_12310420":"43498","customfield_12312320":null,"customfield_12310222":"10002_*:*_1_*:*_1733118611_*|*_1_*:*_1_*:*_1836501_*|*_5_*:*_1_*:*_0","customfield_12312321":null,"resolutiondate":"2011-08-23T05:37:19.655+0000","workratio":-1,"customfield_12312328":null,"customfield_12312329":null,"customfield_12312923":null,"customfield_12312326":null,"customfield_12312920":null,"customfield_12310300":null,"customfield_12312327":null,"customfield_12312921":null,"customfield_12312324":null,"customfield_12312720":null,"customfield_12312325":null,"lastViewed":null,"watches":{"self":"https://issues.apache.org/jira/rest/api/2/issue/HIVE-2338/watchers","watchCount":1,"isWatching":false},"created":"2011-08-03T03:41:24.642+0000","customfield_12310192":null,"customfield_12310191":[{"self":"https://issues.apache.org/jira/rest/api/2/customFieldOption/10343","value":"Reviewed","id":"10343"}],"priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/4","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/minor.svg","name":"Minor","id":"4"},"labels":[],"customfield_12312333":null,"customfield_12310230":null,"customfield_12312334":null,"customfield_12313422":"false","customfield_12310310":"2.0","customfield_12312331":null,"customfield_12312332":null,"timeestimate":null,"aggregatetimeoriginalestimate":null,"customfield_12311120":null,"customfield_12312330":null,"versions":[],"issuelinks":[],"customfield_12312339":null,"assignee":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=sohanjain","name":"sohanjain","key":"sohanjain","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Sohan Jain","active":true,"timeZone":"Etc/UTC"},"customfield_12312337":null,"customfield_12312338":null,"updated":"2011-12-16T23:56:50.732+0000","customfield_12312335":null,"customfield_12312336":null,"status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are not closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Done"}},"components":[{"self":"https://issues.apache.org/jira/rest/api/2/component/12314270","id":"12314270","name":"Diagnosability","description":"Error messages, logging, diagnostics, etc."},{"self":"https://issues.apache.org/jira/rest/api/2/component/12312584","id":"12312584","name":"Metastore","description":"Tracks issue dealing with metastore."}],"timeoriginalestimate":null,"description":"Every failure in an alter table function always return a MetaException. When altering tables and catching exceptions, we throw a MetaException in the \"finally\" part of a try-catch-finally block, which overrides any other exceptions thrown.","customfield_10010":null,"timetracking":{},"customfield_12312026":null,"customfield_12312023":null,"customfield_12310320":null,"customfield_12312024":null,"customfield_12312340":null,"attachment":[{"self":"https://issues.apache.org/jira/rest/api/2/attachment/12489155","id":"12489155","filename":"HIVE-2338.1.patch","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=sohanjain","name":"sohanjain","key":"sohanjain","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Sohan Jain","active":true,"timeZone":"Etc/UTC"},"created":"2011-08-03T04:11:56.115+0000","size":921,"mimeType":"text/plain","content":"https://issues.apache.org/jira/secure/attachment/12489155/HIVE-2338.1.patch"},{"self":"https://issues.apache.org/jira/rest/api/2/attachment/12489280","id":"12489280","filename":"HIVE-2338.2.patch","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=sohanjain","name":"sohanjain","key":"sohanjain","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Sohan Jain","active":true,"timeZone":"Etc/UTC"},"created":"2011-08-03T23:59:25.644+0000","size":793,"mimeType":"text/plain","content":"https://issues.apache.org/jira/secure/attachment/12489280/HIVE-2338.2.patch"}],"aggregatetimeestimate":null,"customfield_12312341":null,"customfield_12312220":null,"customfield_12312022":null,"customfield_12310921":null,"customfield_12310920":"123820","customfield_12312823":null,"summary":"Alter table always throws an unhelpful error on failure","creator":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=sohanjain","name":"sohanjain","key":"sohanjain","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Sohan Jain","active":true,"timeZone":"Etc/UTC"},"subtasks":[],"customfield_12310291":null,"reporter":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=sohanjain","name":"sohanjain","key":"sohanjain","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Sohan Jain","active":true,"timeZone":"Etc/UTC"},"customfield_12310290":null,"aggregateprogress":{"progress":0,"total":0},"customfield_12311024":null,"environment":null,"customfield_12313520":null,"customfield_12311020":null,"duedate":null,"customfield_12310250":null,"progress":{"progress":0,"total":0},"comment":{"comments":[{"self":"https://issues.apache.org/jira/rest/api/2/issue/12517744/comment/13078572","id":"13078572","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=jiraposter%40reviews.apache.org","name":"jiraposter@reviews.apache.org","key":"jiraposter@reviews.apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"jiraposter@reviews.apache.org","active":true,"timeZone":"Etc/UTC"},"body":"\n-----------------------------------------------------------\nThis is an automatically generated e-mail. To reply, visit:\nhttps://reviews.apache.org/r/1276/\n-----------------------------------------------------------\n\nReview request for hive and Paul Yang.\n\n\nSummary\n-------\n\nEvery failure in an alter table function always return a MetaException. When altering tables and catching exceptions, we throw a MetaException in the \"finally\" part of a try-catch-finally block, which overrides any other exceptions thrown.\n\nThis patch moves the code to throw an error on a commit failure outside of the try-catch-finally block.  So any errors within the try block are thrown properly.  Only when success == false will we throw a MetaException.\n\n\nThis addresses bug HIVE-2338.\n    https://issues.apache.org/jira/browse/HIVE-2338\n\n\nDiffs\n-----\n\n  trunk/metastore/src/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java 1152020 \n\nDiff: https://reviews.apache.org/r/1276/diff\n\n\nTesting\n-------\n\nTested failing alter table metastore commands; they threw InvalidOperationExceptions as they should, instead of meaningless MetaExceptions.\n\n\nThanks,\n\nSohan\n\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=jiraposter%40reviews.apache.org","name":"jiraposter@reviews.apache.org","key":"jiraposter@reviews.apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"jiraposter@reviews.apache.org","active":true,"timeZone":"Etc/UTC"},"created":"2011-08-03T04:13:27.623+0000","updated":"2011-08-03T04:13:27.623+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12517744/comment/13079127","id":"13079127","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=jghoman","name":"jghoman","key":"jghoman","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=jghoman&avatarId=10318","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=jghoman&avatarId=10318","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=jghoman&avatarId=10318","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=jghoman&avatarId=10318"},"displayName":"Jakob Homan","active":true,"timeZone":"America/Los_Angeles"},"body":"+1 with the nit that the comment about that code being reached only if the try-catch-finally isn't triggered isn't particularly helpful since it describes the obvious path through the code and won't be interesting once this patch goes in.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=jghoman","name":"jghoman","key":"jghoman","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=jghoman&avatarId=10318","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=jghoman&avatarId=10318","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=jghoman&avatarId=10318","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=jghoman&avatarId=10318"},"displayName":"Jakob Homan","active":true,"timeZone":"America/Los_Angeles"},"created":"2011-08-03T23:54:26.686+0000","updated":"2011-08-03T23:54:26.686+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12517744/comment/13079130","id":"13079130","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=jiraposter%40reviews.apache.org","name":"jiraposter@reviews.apache.org","key":"jiraposter@reviews.apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"jiraposter@reviews.apache.org","active":true,"timeZone":"Etc/UTC"},"body":"\n-----------------------------------------------------------\nThis is an automatically generated e-mail. To reply, visit:\nhttps://reviews.apache.org/r/1276/\n-----------------------------------------------------------\n\n(Updated 2011-08-03 23:59:06.774216)\n\n\nReview request for hive and Paul Yang.\n\n\nChanges\n-------\n\n-remove unecessary comment\n\n\nSummary\n-------\n\nEvery failure in an alter table function always return a MetaException. When altering tables and catching exceptions, we throw a MetaException in the \"finally\" part of a try-catch-finally block, which overrides any other exceptions thrown.\n\nThis patch moves the code to throw an error on a commit failure outside of the try-catch-finally block.  So any errors within the try block are thrown properly.  Only when success == false will we throw a MetaException.\n\n\nThis addresses bug HIVE-2338.\n    https://issues.apache.org/jira/browse/HIVE-2338\n\n\nDiffs (updated)\n-----\n\n  trunk/metastore/src/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java 1152020 \n\nDiff: https://reviews.apache.org/r/1276/diff\n\n\nTesting\n-------\n\nTested failing alter table metastore commands; they threw InvalidOperationExceptions as they should, instead of meaningless MetaExceptions.\n\n\nThanks,\n\nSohan\n\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=jiraposter%40reviews.apache.org","name":"jiraposter@reviews.apache.org","key":"jiraposter@reviews.apache.org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"jiraposter@reviews.apache.org","active":true,"timeZone":"Etc/UTC"},"created":"2011-08-03T23:59:27.041+0000","updated":"2011-08-03T23:59:27.041+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12517744/comment/13089279","id":"13089279","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=cwsteinbach","name":"cwsteinbach","key":"cwsteinbach","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Carl Steinbach","active":true,"timeZone":"America/Los_Angeles"},"body":"Committed to trunk. Thanks Sohan!","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=cwsteinbach","name":"cwsteinbach","key":"cwsteinbach","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Carl Steinbach","active":true,"timeZone":"America/Los_Angeles"},"created":"2011-08-23T05:37:19.687+0000","updated":"2011-08-23T05:37:19.687+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12517744/comment/13089422","id":"13089422","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=hudson","name":"hudson","key":"hudson","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Hudson","active":true,"timeZone":"Etc/UTC"},"body":"Integrated in Hive-trunk-h0.21 #910 (See [https://builds.apache.org/job/Hive-trunk-h0.21/910/])\n    HIVE-2338. Alter table always throws an unhelpful error on failure (Sohan Jain via cws)\n\ncws : http://svn.apache.org/viewcvs.cgi/?root=Apache-SVN&view=rev&rev=1160551\nFiles : \n* /hive/trunk/metastore/src/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=hudson","name":"hudson","key":"hudson","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Hudson","active":true,"timeZone":"Etc/UTC"},"created":"2011-08-23T12:33:23.002+0000","updated":"2011-08-23T12:33:23.002+0000"}],"maxResults":5,"total":5,"startAt":0},"votes":{"self":"https://issues.apache.org/jira/rest/api/2/issue/HIVE-2338/votes","votes":0,"hasVoted":false},"worklog":{"startAt":0,"maxResults":20,"total":0,"worklogs":[]},"customfield_12311820":"0|i0ljkf:"}}